core/local/change.js

```diff
@@ -193,35 +197,76 @@ function isChildAdd (a /*: LocalChange */, b /*: LocalChange */) { return childO
 // $FlowFixMe
 function toString (a /*: LocalChange */) /*: string */ { return '(' + a.type + ': ' + (a.old && a.old.path) + '-->' + a.path + ')' }
 
-function fromEvent (e/*: LocalEvent */) /*: LocalChange */ {
-  const change = _fromEvent(e)
-  log.debug(_.pick(change, ['path', 'ino', 'wip']), `${e.type} -> ${change.type}`)
-  for (let prop of ['md5sum', 'old', 'wip']) {
-    // $FlowFixMe
-    if (change[prop] == null) delete change[prop]
+function dirAddition (e /*: LocalDirAdded */) /*: LocalDirAddition */ {
+  log.debug({path: e.path}, 'addDir = DirAddition')
+  const change /*: LocalDirAddition */ = {
+    sideName,
+    type: 'DirAddition',
+    path: e.path,
+    stats: e.stats,
+    ino: e.stats.ino
+  }
+  if (e.old) change.old = e.old
+  if (e.wip) change.wip = e.wip
```

Comment:
By curiosity, we can have `wip` on directory? In which conditions?

Reply:
Currently `wip` is set on an `addDir` event in case the directory doesn't exist anymore (although this may not be such a good idea).